woocommerce/woocommerce
# Optimized `wireit` Fingerprinting File Inputs

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Previously we were making assumptions about what files could be considered "build outputs" in cases where a `package.json` file did not explicitly define `"files"`. In practice, this ended up causing problems because some source file changes would still be cause and cause downstream packages to start building before the dependency was finished building. Rather than relying on this assumption we should instead just explicitly define the build outputs for all packages in the monorepo.

This pull request makes changes to the `.pnpmfile.js` hook so that it only includes explicitly defined output files. I also ran that change against all of the `package.json` files as well as fixed a slight bug that was causing some linked packages to be missed.

Closes #42011.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Run `pnpm --filter='@woocommerce/plugin-woocommerce' build`
2. Run the command again and confirm that it runs significantly faster.
3. Make a change in a `plugins/woocommerce/client/legacy/css` file that will change the build output.
4. Re-run the above and confirm that the change is reflected in `plugins/woocommerce/client/assets/css`.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
